polars/polars-core/src/chunked_array/kernels/strings.rs

```diff
@@ -1,7 +1,17 @@
 use crate::prelude::Arc;
-use arrow::array::{ArrayRef, LargeStringArray, UInt32Array};
+use arrow::array::{ArrayRef, UInt32Array, Utf8Array};
+use arrow::buffer::Buffer;
+use arrow::datatypes::DataType;
 
-pub(crate) fn string_lengths(array: &LargeStringArray) -> ArrayRef {
-    let array: UInt32Array = array.iter().map(|v| v.map(|v| v.len() as u32)).collect();
+pub(crate) fn string_lengths(array: &Utf8Array<i64>) -> ArrayRef {
+    let values = array
+        .offsets()
+        .iter()
+        .windows(2)
+        .map(|x| (x[1] - x[0]) as u32);
+
+    let values = Buffer::from_trusted_len_iter(values);
+
+    UInt32Array::from_data(DataType::UInt32, values, array.validity().clone());
```

Comment:
Arrow2' `Utf8Array<O>` has three main methods: 

* `.validity() -> &Option<Bitmap>`
* `.values() -> &Buffer<u8>`
* `.offset() -> &Buffer<O>`

`Buffer` can be deffered to a slice of its type, and is thus easily used for all kinds of iterator operations. This is an example, where we iterate over the offsets to compute the lengths as `u32`, create a new buffer from it, and a new UInt32Array with them. 

This should be about 50-90% faster than the current code, since there is no iteration over the validity nor values.

Reply:
And very idiomatic Rust. Kudo's on this API.